### PR TITLE
Feature/immutable task descriptions and deal parameters

### PR DESCRIPTION
**Pull request by jbern0rd**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@ All notable changes to this project will be documented in this file.
 ### New Features
 - Call `IexecHubContract#viewDeal` instead of legacy ABI in `IexecHubAbstractService#getChainDeal`. (#24)
 - Representations of on-chain object are now immutable. (#24 #25 #26 #30)
+- `TaskDescription` and `DealParams` instances are immutable. (#34)
```
Comment (mcornaton) [inline CHANGELOG.md:10]: Merge with previous line?
```suggestion
- Representations of on-chain object are now immutable. (#24 #25 #26 #34)
```

Comment (jbern0rd) [reply CHANGELOG.md:10]: I don't really know, they are not really on-chain objects representations (comparatively to ChainDeal, ChainTask, ChainApp, ...). Let me think a bit about this.

Comment (jbern0rd) [reply CHANGELOG.md:10]: Or maybe
```suggestion
- Representations of on-chain and off-chain objects are now immutable. (#24 #25 #26 #30 #34)
```

Comment (mcornaton) [reply CHANGELOG.md:10]: Perfect!

**Review by mcornaton — APPROVED** [review]

**Pull request merged**